src/drivers/nuttx/lv_nuttx_cache.c

```diff
@@ -44,28 +45,44 @@ void lv_nuttx_cache_init(void)
 {
     lv_draw_buf_handlers_t * handlers = lv_draw_buf_get_handlers();
     handlers->invalidate_cache_cb = invalidate_cache;
+    handlers->flush_cache_cb = flush_cache;
 }
 
 /**********************
  *   STATIC FUNCTIONS
  **********************/
 
-static void invalidate_cache(const lv_draw_buf_t * draw_buf, const lv_area_t * area)
+static void draw_buf_to_region(
+    const lv_draw_buf_t * draw_buf, const lv_area_t * area,
+    lv_uintptr_t * start, lv_uintptr_t * end)
 {
     LV_ASSERT_NULL(draw_buf);
+    LV_ASSERT_NULL(area);
```

Comment:
Set `area` to NULL is a valid parameter to indicate operate on the whole draw buffer.

```c
/**
 * Invalidate the cache of the buffer
 * @param draw_buf     the draw buffer needs to be invalidated
 * @param area         the area to invalidate in the buffer,
 *                     use NULL to invalidate the whole draw buffer address range
 */
void lv_draw_buf_invalidate_cache(const lv_draw_buf_t * draw_buf, const lv_area_t * area);
```

Reply:
The `area == NULL` situation has been handled in function: `lv_draw_buf_invalidate_cache_user` `lv_draw_buf_flush_cache_user`

```c
lv_area_t full;
if(area == NULL) {
    draw_buf_get_full_area(draw_buf, &full);
    area = &full;
}
```